### PR TITLE
CATALOG-2408 - Bulk Pricing updates per-variant

### PR DESCRIPTION
**Pull request by bookernath**
#### What?

Price Lists support different bulk pricing tiers for each variant, so we need to update the viewModel whenever these tiers change.

#### Tickets / Documentation

- [CATALOG-2408](https://jira.bigcommerce.com/browse/CATALOG-2408)

#### Screenshots (if appropriate)

GIF

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "author": "BigCommerce",
   "license": "MIT",
   "dependencies": {
-    "@bigcommerce/stencil-utils": "^1.0.10",
+    "@bigcommerce/stencil-utils": "bookernath/stencil-utils#d263f2ef03aa88e1361c8b4331204e4e4e8bd5e4",
```
Comment (bookernath) [inline package.json]: this gets replaced with a new version once the utils PR is merged.

File: assets/js/theme/cart.js [hunk 2]
```diff
@@ -98,7 +98,7 @@ export default class Cart extends PageManager {
             const $messageBox = $('.alertMessageBox');
             const item = $('[name="item_id"]', $form).attr('value');
 
-            utils.api.productAttributes.optionChange(item, $form.serialize(), (err, result) => {
```
Comment (mjschock) [inline assets/js/theme/cart.js]: looking at the new `optionChange` api, it seems that we don't need to pass `null` here

File: assets/js/theme/common/product-details.js [hunk 3]
```diff
@@ -107,11 +108,11 @@ export default class ProductDetails {
             return;
         }
 
-        utils.api.productAttributes.optionChange(productId, $form.serialize(), (err, response) => {
+        utils.api.productAttributes.optionChange(productId, $form.serialize(), 'products/bulk-discount-rates', (err, response) => {
             const productAttributesData = response.data || {};
-
+            const bulkPricingHtml = response.content || {};
```
Comment (mjschock) [inline assets/js/theme/common/product-details.js]: could we match the productAttributesData here, like so:

```
const content = response.content || {};
```
...
```
updateView(data, content)
```
etc.

File: assets/js/theme/common/product-details.js [hunk 4]
```diff
@@ -393,6 +393,13 @@ export default class ProductDetails {
             viewModel.$addToCart.prop('disabled', false);
             viewModel.$increments.prop('disabled', false);
         }
+
+        // If Bulk Pricing rendered HTML is available
+        if (content) {
```
Comment (mjschock) [inline assets/js/theme/common/product-details.js]: does this work?  i thought it was grabbing a field off of content... like this instead:

```
updateView(data, content) { // content will be {} if response.content is empty above
```
...
```
if (content.bulkPricingContent) { // or whatever it's called
  viewModel.$bulkPricing.html(content.bulkPricingContent)
```

Comment (bookernath) [reply assets/js/theme/common/product-details.js]: Content returns a blank string which evaluates to no if there are no tiers.

**Comment by bookernath** [comment]
@mjschock I think I've addressed most of your feedback if you wanna take a peek. Also rebased to pull in the JQ update.

**Pull request merged**